projectroles/forms.py

```diff
@@ -338,20 +338,48 @@ def _set_app_setting_widget(self, app_name, s_field, s_key, s_val):
                 )
             else:
                 json_data = self.app_settings.get_default(
-                    app_name=app_name, setting_name=s_key
+                    app_name=app_name,
+                    setting_name=s_key,
+                    project=None,
                 )
             self.initial[s_field] = json.dumps(json_data)
         else:
             if s_val.get('options'):
-                self.fields[s_field] = forms.ChoiceField(
-                    choices=[
-                        (int(option), int(option))
-                        if s_val['type'] == 'INTEGER'
-                        else (option, option)
-                        for option in s_val.get('options')
-                    ],
-                    **setting_kwargs
-                )
+                if callable(s_val.get('default')):
+                    if self.instance.pk:
+                        self.initial[s_field] = s_val.get('default')(
+                            project=self.instance
+                        )
+                        self.fields[s_field] = forms.ChoiceField(
+                            choices=[
+                                (str(option), str(option))
+                                for option in s_val.get('options')(
```

Comment:
I was unable to find the check for whether `options` is a callable. There only seems to be that check for `default`? Default and options do not depend on each other and either, neither or both can be set as callable.

Possibly we didn't catch this because your example settings with a callable `option` also have a callable `default`, so this condition is always reached?